### ISSUE TITLE
idna correctly returns bytes which breaks twisted.internet._sslverify._idnaBytes on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7287]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)| @hynek reported|
|-|-|
|Trac ID|trac#7287|
|Type|enhancement|
|Created|2014-05-04 16:31:13Z|
|Branch|https://github.com/twisted/twisted/tree/idnabytes-7287|

If I’m not mistaken, we just have to remove the ``.encode("ascii")`` from https://github.com/twisted/twisted/blob/ad10b44f2355129bd839ab2cd2080c44751ad877/twisted/internet/_sslverify.py#L86 .

Not sure if this is important enough to become a regression/blocker for 14.0.

<details><summary>Searchable metadata</summary>

```
trac-id__7287 7287
type__enhancement enhancement
reporter__hynek hynek
priority__highest highest
milestone__Python_3_x Python-3.x
branch__branches_idnabytes_7287 branches/idnabytes-7287
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1399221073000000 1399221073000000
changetime__1430752808531822 1430752808531822
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

Can't we just use Python's built-in idna code on Python 3?

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

This makes the new python 3.4 bot fail: https://buildbot.twistedmatrix.com/builders/debian7-py3.4/builds/0/steps/shell/logs/stdio

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Also worth noting: the built-in idna encoder is the 2003-era standard, and the PyPI package supports the newer 2008 RFC, so we should use the PyPI one instead, I think.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

(In [44645]) Branching to idnabytes-7287.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl commented|
|-|-|

Builders spun. Please see the unsupported list, with the now-green debian7-py33 builder. (https://buildbot.twistedmatrix.com/boxes-unsupported?branch=/branches/idnabytes-7287)

This also changes Twisted to rely on versions of idna that aren't broken (0.6 and above).

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_7"></a>|@hynek set owner to @hawkowl|
|-|-|

Needs a topfile. Ready for merge otherwise.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44649]) Merge idnabytes-7287: Fix twisted.internet._sslverify._idnaBytes on Python 3

Author: hawkowl
Reviewer: hynek
Fixes: [#7287](https://github.com/twisted/twisted/issues/7287)
